js/interface.js

```diff
@@ -1426,6 +1426,10 @@ $('#show-access-rules').click(function() {
   $tbody.html('');
   $accessRulesList.css('opacity', 0.5);
 
+  if (!Array.isArray(currentDataSourceRules)) {
```

Comment:
as I have mentioned previously, if the data sources is not an array it means "default access". It looks like you have coded it to be "no access", which is incorrect. Please explain? Did you read my previous explanation?

Reply:
@squallstar Sorry, yes, I read your comment and I think I understood it the other way around, I will correct the logic and as I understand this change will not only be in the data-source component